src/utils/componentLifecycleDecorator.js

```diff
@@ -14,18 +14,30 @@ export default function componentLifecycleDecorator ({registerEvents, instanceMe
       this._unregisterEvents = null;
     }
 
+    // Stash component's own lifecycle methods to be invoked later
+    const componentDidMount = Component.prototype.hasOwnProperty("componentDidMount") ? Component.prototype.componentDidMount : null;
```

Comment:
Actually the more I think about it, it's cleaner to just save the whole original prototype rather than individual parts of it.